File: Tutorial1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/MyApplication.java
package com.example.tutorial1daggerbasics;

import android.app.Application;

import com.example.tutorial1daggerbasics.di.DaggerMyExampleComponent;
import com.example.tutorial1daggerbasics.di.MyExampleComponent;
import com.example.tutorial1daggerbasics.di.MyExampleModule;

public class MyApplication extends Application {
    private MyExampleComponent mMyComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        mMyComponent = createMyComponent();
    }

    MyExampleComponent getMyComponent() {
        return mMyComponent;
    }

    private MyExampleComponent createMyComponent() {
        return DaggerMyExampleComponent.builder()
                .myExampleModule(new MyExampleModule())
                .build();
    }
}

File: Tutorial4-6SubComponentBuilders-MultipleModulesAndActvities/src/main/java/com/example/tutorial4_6subcomponentbuilders_multiplemodulesandactvities/MyApplication.kt
package com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities

import android.app.Application
import com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.di.ApplicationComponent
import com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.di.DaggerApplicationComponent

class MyApplication : Application() {
   lateinit var applicationComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()
        applicationComponent = DaggerApplicationComponent.builder().application(this).build()
    }

}